packages/base/src/utils.ts

```diff
@@ -1,54 +1,78 @@
-const blake2b = require("blake2b");
-const isEqual = require("lodash.isequal");
-const { xxHash32 } = require("js-xxhash");
-const { bytes, number } = require("@ckb-lumos/codec");
-const { BI } = require("@ckb-lumos/bi");
-const blockchain = require("./blockchain");
+import blake2b, { Blake2b } from "blake2b";
+import isEqual from "lodash.isequal";
+import { xxHash32 } from "js-xxhash";
+import { bytes, number, BytesLike } from "@ckb-lumos/codec";
+import { BI, BIish } from "@ckb-lumos/bi";
+import * as blockchain from "./blockchain";
+import { Script, Input } from "./api";
+import { Hash, HexNumber, HexString } from "./primitive";
+
+type CKBHasherOptions = {
+  outLength?: number;
+};
 
 const { bytify, hexify, bytifyRawString } = bytes;
 class CKBHasher {
-  constructor(options = {}) {
+  hasher: Blake2b;
+  outLength: number;
+
+  constructor(options: CKBHasherOptions = {}) {
     const { outLength = 32 } = options;
     this.outLength = outLength;
     this.hasher = blake2b(
       outLength,
-      null,
-      null,
+      undefined,
+      undefined,
       bytifyRawString("ckb-default-hash")
     );
   }
 
-  update(data) {
+  update(data: string | ArrayBuffer): this {
     this.hasher.update(bytify(data));
     return this;
   }
 
-  digestHex() {
+  digestHex(): Hash {
     const out = new Uint8Array(this.outLength);
     this.hasher.digest(out);
     return hexify(out.buffer);
   }
 }
 
-function ckbHash(data) {
+function ckbHash(data: BytesLike): Hash {
   const hasher = new CKBHasher();
   hasher.update(bytes.bytify(data));
   return hasher.digestHex();
 }
 
-function computeScriptHash(script) {
+/**
+ * compute lock/type hash
+ *
+ * @param script
+ * @param _options @deprecated this option has no effect
+ */
+function computeScriptHash(
+  script: Script,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
```suggestion
```

Reply:
Seems no changes?
![image](https://user-images.githubusercontent.com/20639676/198239336-e93cf7f2-9aab-431f-8842-aaa239556716.png)

Reply:
the `// no-unused-vars` is unnecessary here